### ISSUE TITLE
Accepts incorrect tag_num or tag_type

### ISSUE THREAD

**Issue by KevinSiwicke-NOAA** [tristan-sebens/metl#82]
I accidentally mistyped a tag num, and it still uploaded the data with the incorrect number. 11853 was accepted but 118353 was correct, and it also was accepted.

Similarly, I tried two different tag_types ('ET' and 'RE') for the same tag, and both were accepted.

It does not appear to be cross referencing the recovery table.

**Comment by CindyTribuzio-NOAA** [comment]
Did the same test with the same result. However, it would not accept nonsense tag_type (e.g., "XE"). It would accept nonsense tag_num.

**Comment by CindyTribuzio-NOAA** [comment]
Highlighting this as a critical accuracy check. Tags must be in release and recovery table before being uploaded.

**Comment by tristan-sebens** [comment]
Incorrect tag numbers should be rejected by the DB based on the constraints placed on the metadata table (TAG). 

I'll double check that these constraints are correctly specified.

**Comment by tristan-sebens** [comment]
I've reviewed the DB table and confirmed that the necessary constraints are in place. However, I believe I have identified a separate cause of this issue. I believe that the cause of this issue lies in the SEQ_NUM and SPECIES_CODE fields. 

The constraints on the TAG table say that a record may only be entered into the table if the record matches the primary keys of records in both the RECOVERY and RELEASE tables. However, in both tables the primary key is a composite made up of four fields: TAG_NUM, TAG_TYPE, REL/REC_SEQ, and SPECIES CODE. Only the first two of these are currently present in the output FieldMap for the TAG table, so the other two are inserted as NULL values. Oracle documentation states that the DB engine cannot compare partially NULL composite primary keys, so the constraint check fails.

Once again, this is an issue which should be addressed by the newest version once released. This version includes both of the missing fields (SEQ_NUM and SPECIES_CODE) in the output FieldMap, and so the users will be able to supply values for these fields. Once each field has a value, the table constraint should be able to be evaluated correctly, and new TAG records which do not match records in the RECOVERY and RELEASE tables should be successfully rejected.
